src/auth.ts

```diff
@@ -702,8 +702,8 @@ async function refreshTokens(
   }
 }
 
-export async function getAccessToken(refreshToken: string, authScopes: string[]) {
-  if (haveValidTokens(refreshToken, authScopes)) {
+export async function getAccessToken(refreshToken: string, authScopes: string[]): Promise<Tokens> {
```

Comment:
The changes to auth.ts and apiv2.ts are somewhat unrelated clean up - i spent a bunch of time poking around these files trying to understand the issue, so I did some tidying up while i was there.